gcp/src/test/java/org/apache/iceberg/gcp/biglake/BigLakeCatalogTest.java

```diff
@@ -0,0 +1,385 @@
+            GCPProperties.BIGLAKE_PROJECT_ID,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation);
+
+    channelProvider = mockServiceHelper.createChannelProvider();
+    MetastoreServiceSettings settings =
+        MetastoreServiceSettings.newBuilder()
+            .setTransportChannelProvider(channelProvider)
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .build();
+
+    bigLakeCatalogUsingMockService = new BigLakeCatalog();
+    bigLakeCatalogUsingMockService.setConf(new Configuration());
+    bigLakeCatalogUsingMockService.initialize(
+        CATALOG_ID,
+        properties,
+        GCP_PROJECT,
+        GCP_REGION,
+        new BigLakeClient(settings, GCP_PROJECT, GCP_REGION));
+
+    mockBigLakeClient = mock(BigLakeClient.class);
+    bigLakeCatalogUsingMockClient = new BigLakeCatalog();
+    bigLakeCatalogUsingMockClient.setConf(new Configuration());
+    bigLakeCatalogUsingMockClient.initialize(
+        CATALOG_ID, properties, GCP_PROJECT, GCP_REGION, mockBigLakeClient);
+  }
+
+  @AfterEach
+  public void after() {
+    mockServiceHelper.stop();
+  }
+
+  @Override
+  protected boolean requiresNamespaceCreate() {
+    return true;
+  }
+
+  @Override
+  protected BigLakeCatalog catalog() {
+    return bigLakeCatalogUsingMockService;
+  }
+
+  @Override
+  protected boolean supportsNamesWithSlashes() {
+    return false;
+  }
+
+  @Test
+  public void testDefaultWarehouseWithDatabaseLocation_asExpected() {
+    when(mockBigLakeClient.getDatabase(DatabaseName.of(GCP_PROJECT, GCP_REGION, CATALOG_ID, "db")))
+        .thenReturn(
+            Database.newBuilder()
+                .setHiveOptions(HiveDatabaseOptions.newBuilder().setLocationUri("db_folder"))
+                .build());
+
+    assertEquals(
+        "db_folder/table",
+        bigLakeCatalogUsingMockClient.defaultWarehouseLocation(TableIdentifier.of("db", "table")));
+  }
+
+  @Test
+  public void testDefaultWarehouseWithoutDatabaseLocation_asExpected() {
+    when(mockBigLakeClient.getDatabase(DatabaseName.of(GCP_PROJECT, "us", CATALOG_ID, "db")))
+        .thenReturn(
+            Database.newBuilder().setHiveOptions(HiveDatabaseOptions.getDefaultInstance()).build());
+
+    assertEquals(
+        warehouseLocation + "/db.db/table",
+        bigLakeCatalogUsingMockClient.defaultWarehouseLocation(TableIdentifier.of("db", "table")));
+  }
+
+  @Test
+  public void testRenameTable_differentDatabase_fail() {
+    Exception exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () ->
+                bigLakeCatalogUsingMockClient.renameTable(
+                    TableIdentifier.of("db0", "t1"), TableIdentifier.of("db1", "t2")));
+    assertEquals(
+        "Cannot rename table db0.t1 to db1.t2: database must match", exception.getMessage());
+  }
+
+  @Test
+  public void testCreateNamespace_createCatalogWhenEmptyNamespace() throws Exception {
```

Comment:
I believe test methods shouldn't contain `_` and rather use camelCase

Reply:
Done, changed all.

Reply:
there seem to be still a few left, like `testDefaultWarehouseWithDatabaseLocation_asExpected` and some others

Reply:
Sorry I missed some because of starting the cleanup in middle. Done.